src/app/qgisapp.cpp

```diff
@@ -1890,10 +1892,56 @@ QgisApp::QgisApp( QSplashScreen *splash, bool restorePlugins, bool skipBadLayers
   mCodeEditorWidgetFactory.reset( std::make_unique< QgsCodeEditorOptionsFactory >() );
   mBabelGpsDevicesWidgetFactory.reset( std::make_unique< QgsGpsDeviceOptionsFactory >() );
   mCustomProjectionsWidgetFactory.reset( std::make_unique< QgsCustomProjectionOptionsFactory >() );
+  mFontOptionsWidgetFactory.reset( std::make_unique< QgsFontOptionsFactory >() );
 
 #ifdef HAVE_3D
   m3DOptionsWidgetFactory.reset( std::make_unique< Qgs3DOptionsFactory >() );
 #endif
+
+  connect( QgsApplication::fontManager(), &QgsFontManager::fontDownloaded, this, [ = ]( const QStringList & families, const QString & licenseDetails )
+  {
+    const QString shortMessage = tr( "Installed font %1" ).arg( families.join( QStringLiteral( ", " ) ) );
+    QgsMessageBarItem *messageWidget = QgsMessageBar::createMessage( QString(), shortMessage );
+    if ( !licenseDetails.isEmpty() )
+    {
+      QPushButton *detailsButton = new QPushButton( tr( "View License" ) );
+      connect( detailsButton, &QPushButton::clicked, this, [detailsButton, licenseDetails]
+      {
+        QgsMessageViewer *dialog = new QgsMessageViewer( detailsButton );
+        dialog->setTitle( tr( "Font License" ) );
+        dialog->setMessage( licenseDetails, QgsMessageOutput::MessageText );
+        dialog->showMessage();
+      } );
+      messageWidget->layout()->addWidget( detailsButton );
+    }
+    messageBar()->pushWidget( messageWidget, Qgis::MessageLevel::Info, 0 );
+
+    // refresh canvas to get proper rendering using that font
+    const QList< QgsMapCanvas * > canvases = mapCanvases();
+    for ( QgsMapCanvas *canvas : canvases )
+      canvas->refreshAllLayers();
```

Comment:
I guess there's no way to be cleverer about it (i.e. only triggering layer fresh for those that actually rely on the downloaded font) as a new font would mean new metrics, possibly much smaller or larger label size, etc. OK.

(sorry, just thinking this through out-loud ;) )

Reply:
I don't think it's a huge issue really -- this will occur only when a new font is downloaded, so a user may see this happen at most a couple of times during their life 😁